source/commands/env/export/components/hooks/useExport.ts

```diff
@@ -0,0 +1,70 @@
+import { useState } from 'react';
+import { PermitSDK } from './PermitSDK.js';
+import { ExportState } from '../../types.js';
+import { createWarningCollector, generateProviderBlock } from '../../utils.js';
+import { ResourceGenerator } from '../../generators/ResourceGenerator.js';
+import { RoleGenerator } from '../../generators/RoleGenerator.js';
+import { UserAttributesGenerator } from '../../generators/UserAttributesGenerator.js';
+import { RelationGenerator } from '../../generators/RelationGenerator.js';
+import { ConditionSetGenerator } from '../../generators/ConditionSetGenerator.js';
+
+// Define a type for the `scope` parameter
+interface ExportScope {
+	environment_id?: string;
+	project_id?: string;
+	organization_id?: string;
+}
+
+export const useExport = (apiKey: string) => {
+	const [state, setState] = useState<ExportState>({
+		status: '',
+		isComplete: false,
+		error: null,
+		warnings: [],
+	});
+
+	const permit = PermitSDK(apiKey);
+
+	const exportConfig = async (scope: ExportScope) => {
+		try {
+			const warningCollector = createWarningCollector();
+
+			let hcl = `# Generated by Permit CLI
```

Comment:
Please move all the `.hcl` code to static `.hcl` files in a dedicated folder and import them to the code where needed. 
It will:
1. Made the code more readable and keep the indentation
2. Ensure reusable of code snippets

Reply:
It is up to you to decide on the import way of this text and use the templates in it